### PR TITLE
Update sort

### PR DESCRIPTION
**Pull request by YShu7**
add sort system test, ignored now because of incomplete command highlight
update UG
update TagComparator

### CODE REVIEW

File: src/test/java/seedu/saveit/logic/commands/SortCommandTest.java [hunk 1]
```diff
@@ -76,10 +80,18 @@ public void execute_sortAfterUpdate_success() {
...
         SortCommand command = new SortCommand(issueSort);
         assertCommandSuccess(command, model, commandHistory, expectedMessage, expectedModel);
-        assertEquals(Arrays.asList(DANIEL, BENSON, ALICE, issue, GEORGE, FIONA, ELLE, CARL),
+        assertEquals(Arrays.asList(DANIEL, BENSON, ALICE, CARL, ELLE, FIONA, GEORGE, issue),
                 model.getFilteredAndSortedIssueList());
     }
 
+    @Test
+    public void execute_notUnderRootLevel_failure() {
+        model.resetDirectory(Directory.formDirectory(INDEX_FIRST_ISSUE.getOneBased(), 0));
```
Comment (leo-1997) [inline src/test/java/seedu/saveit/logic/commands/SortCommandTest.java]: I think maybe use a new class (e.g. DirectoryBuilder) in test util will be more unified

File: src/test/java/systemtests/SortCommandSystemTest.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    @Test
+    @Ignore
+    public void sort() {
+        Model expectedModel = getModel();
+
+        /* Case: sort issues by tag sort in saveit book, command with leading spaces and trailing spaces */
+        String command = "   " + SortCommand.COMMAND_WORD + " " + TAG_SORT + "   ";
+        ModelHelper.setSortedList(expectedModel, DANIEL, BENSON, ALICE, CARL, ELLE, FIONA, GEORGE);
+        assertCommandSuccess(command, IssueSort.TAG, expectedModel);
+        assertSelectedCardUnchanged();
```
Comment (xllliu) [inline src/test/java/systemtests/SortCommandSystemTest.java]: Currently this assertion may not be working as no card will be selected at root directory.

File: src/test/java/seedu/saveit/logic/commands/SortCommandTest.java (hunk 1 above)
Comment (YShu7) [reply src/test/java/seedu/saveit/logic/commands/SortCommandTest.java]: Yep exactly. Once your PR is merged I will updated this one. I noticed you created alr so didn't bother to create lol.

File: src/test/java/systemtests/SortCommandSystemTest.java (hunk 2 above)
Comment (YShu7) [reply src/test/java/systemtests/SortCommandSystemTest.java]: Noted with thanks. will update soon

File: src/test/java/seedu/saveit/logic/commands/SortCommandTest.java (hunk 1 above)
Comment (leo-1997) [reply src/test/java/seedu/saveit/logic/commands/SortCommandTest.java]: okok

**Pull request merged**